### PR TITLE
Replace `lambdify` function with `lambdify_sympde`

### PR DESCRIPTION
**Pull request by anlavandier**
Add a new function `lambdify_sympde` as an improvement over Sympy's `lambdify`. Previous uses of `lambdify` are replaced by `lambdify_sympde`, which correctly handles Numpy arrays broadcasting rules. Further, add array tests to `test_callable_mapping.py`.

### CODE REVIEW

File: sympde/utilities/tests/test_utilities.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    scalar_input_d = f_d(0)
+
+    scalar_input_ia = f_ia(0)
+    scalar_input_da = f_da(0)
+
+    assert np.array_equal(scalar_input_i, 1)
+    assert np.array_equal(scalar_input_d, 0)
+
+    assert np.array_equal(scalar_input_ia, np.eye(2))
+    assert np.array_equal(scalar_input_da, np.full((2, 2), 0))
```
Comment (yguclu) [inline sympde/utilities/tests/test_utilities.py:36]: Suggestion: `np.full((2, 2), 0)` -> `np.zeros((2, 2))`

File: sympde/utilities/utils.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import numpy as np
+import itertools as it
+from sympy import lambdify
+
+def lambdify_sympde(variables, expr):
+    """
+    Custom lambify function that covers the
+    shortcomings of sympy's lambdify.
```
Comment (yguclu) [inline sympde/utilities/utils.py:8]: This description should be expanded, e.g. by mentioning that the lambdified function correctly handles Numpy's broadcasting rules.

File: sympde/utilities/utils.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+        def f_vec_v(*XYZ):
+            b = np.broadcast(*XYZ)
+            result = np.zeros(scalar_shape + b.shape)
+            for multi_index in it.product(*tuple(range(s) for s in scalar_shape)):
+                result[multi_index] = scalar_functions[multi_index](*XYZ)
+            return result
+        return f_vec_v
+
+
```
Comment (yguclu) [inline sympde/utilities/utils.py:81]: I am surprised that Codacy does not complain about these empty lines at the end of the file :smile:

**Review by yguclu — CHANGES_REQUESTED** [review]
Excellent work! I have some minor comments

File: sympde/utilities/tests/test_utilities.py (hunk 1 above)
Comment (anlavandier) [reply sympde/utilities/tests/test_utilities.py]: Fixed in 335552c

File: sympde/utilities/utils.py (hunk 2 above)
Comment (anlavandier) [reply sympde/utilities/utils.py]: Fixed in 335552c

File: sympde/utilities/tests/test_utilities.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+    array_input_i = f_i(np.linspace(0, 1, 10))
+    array_input_d = f_d(np.linspace(0, 1, 10))
+
+    array_input_ia = f_ia(np.linspace(0, 1, 10))
+    array_input_da = f_da(np.linspace(0, 1, 10))
+
+    assert np.array_equal(array_input_i, np.full((10,), 1))
```
[the remark this answers is not in the document]
Comment (anlavandier) [reply sympde/utilities/tests/test_utilities.py]: Fixed in 335552c

**Comment by anlavandier** [comment]
> Please update the version of SymPDE, so that we can upload it to PyPI and then use it in Psydac.

Changed it from `0.13.1` to `0.14.0` in [3f592f6](https://github.com/pyccel/sympde/pull/103/commits/3f592f61d2f4db01ba93e8a1081fbe960ebd79ed)

**Review by yguclu — APPROVED** [review]

**Pull request merged**